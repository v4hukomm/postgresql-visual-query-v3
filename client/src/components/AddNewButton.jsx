import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';
import * as PropTypes from 'prop-types';
import { Button } from 'reactstrap';

export function AddNewButton({ id, size, onClick }) {
  return (
    <Button
      className=""
      outline
      color="info"
      size={size}
      id={id}
      onClick={onClick}
    >
      <FontAwesomeIcon icon="plus" />
    </Button>
  );
}

AddNewButton.propTypes = {
  id: PropTypes.string,
  size: PropTypes.string,
  onClick: PropTypes.func,
};
