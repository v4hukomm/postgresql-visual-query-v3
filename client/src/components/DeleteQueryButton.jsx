import React from 'react';
import { Button } from 'reactstrap';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as PropTypes from 'prop-types';
import { deleteQuery, setActiveQuery } from '../actions/queryActions';
import { deleteQueries, removeMainFromQueries } from '../actions/queriesActions';

export const DeleteQueryButton = (props) => {
  const handleOnClick = () => {
    props.deleteQuery();

    const mainQuery = props.queries.find(query => query.id === 0);

    if (mainQuery) {
      props.removeMainFromQueries();
      props.setActiveQuery(mainQuery);
    } else {
      props.deleteQueries();
    }
  };

  return (
    <Button color="danger" className="btn-sm" onClick={handleOnClick} style={{ borderTopLeftRadius: '0px', borderBottomLeftRadius: '0px' }}>
      <FontAwesomeIcon icon="times" />
    </Button>
  );
};

DeleteQueryButton.propTypes = {
  deleteQuery: PropTypes.func,
  deleteQueries: PropTypes.func,
  queries: PropTypes.arrayOf(PropTypes.shape({})),
  removeMainFromQueries: PropTypes.func,
  setActiveQuery: PropTypes.func,
};

const mapStateToProps = store => ({
  queries: store.queries,
});

const mapDispatchToProps = {
  deleteQuery,
  deleteQueries,
  removeMainFromQueries,
  setActiveQuery,
};

export default connect(mapStateToProps, mapDispatchToProps)(DeleteQueryButton);
