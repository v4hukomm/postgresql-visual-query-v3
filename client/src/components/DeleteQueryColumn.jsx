import React, { useState } from 'react';
import { connect } from 'react-redux';
import { 
    Button,
    CustomInput,
    Col,
    Input,
    InputGroup,
    Row, 
} from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { updateColumn, removeColumn } from '../actions/queryActions';

export const DeleteQueryColumn = (props) => {

    const [returning, setReturning] = useState(props.data.returning);
    const [returningOnly, setReturningOnly] = useState(props.data.returning);

    let column = props.data;

    const handleChange = (e) => {
        column.column_filters.map((filter) => {
            if (filter.id === parseInt(e.target.name)) {
                filter.filter = e.target.value;
            }
            return filter;
            });
    };

    const handleSave = () => {
        props.updateColumn(column);
    };

    const handleRemoveColumn = () => {
        props.removeColumn(props.data);
    };

    const handleReturningSwitch = () => {
        column.returning = !column.returning;
        setReturning(current => !current);

        props.updateColumn(column);
    };  

    const handleReturningOnlySwitch = () => {
        column.returningOnly = !column.returningOnly;
        setReturningOnly(current => !current);

        props.updateColumn(column);
    };
    
    return (
        <div>
            <div className='border-bottom border-right pr-2 pl-1 border-dark'>
                        <h6 className='d-inline'>
                            {column.table_name}.{column.column_name}
                        </h6>
                        <Button
                        className="mr-2 float-right"
                        size="sm"
                        color="danger"
                        onClick={handleRemoveColumn}
                        >
                        <FontAwesomeIcon icon="times" />
                        </Button>
                        <InputGroup size="sm">
                        <CustomInput
                         className=""
                         type="switch"
                         id={`column-returning-${column.id}`}
                         key={column.id}
                         name="returning"
                         disabled={props.returning || returningOnly}
                         checked={returning}
                         onChange={handleReturningSwitch}
                         label="Returning">
                        </CustomInput>
                        <CustomInput
                         className=""
                         type="switch"
                         id={`column-returning-only-${column.id}`}
                         key={column.id}
                         name="returning-only"
                         disabled={props.returning}
                         checked={returningOnly}
                         onChange={handleReturningOnlySwitch}
                         label="Returning only">
                        </CustomInput>
                        </InputGroup>
                </div>
                        <Col className='mr-2 border-bottom'>
                            {column.column_filters.map((filter) => (
                            <Row className="pt-2 pb-2 pr-2 pl-1 border-top" key={filter.id}>
                                <Input
                                 id={`column-filter-${filter.id}`}
                                 name={filter.id}
                                 key={filter.id}
                                 type="text"
                                 placeholder="Ex. = 3"
                                 defaultValue={filter.filter}
                                 onBlur={handleSave}
                                 onChange={handleChange}
                                 disabled={returningOnly}>
                               </Input>
                               <p className='ml-2'>AND</p>
                            </Row>
                            ))}
                        </Col>
        </div>
    );
};

const mapStateToProps = (store) => ({
    returning: store.query.returning,
});

const mapDispatchToProps = {
    updateColumn,
    removeColumn,
};


export default connect(mapStateToProps, mapDispatchToProps)(DeleteQueryColumn);