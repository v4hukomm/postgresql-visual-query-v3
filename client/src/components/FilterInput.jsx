import React from 'react';
import { connect } from 'react-redux';
import {
  Input,
} from 'reactstrap';
import { updateColumnFilter } from '../actions/queryActions';

export const FilterInput = (props) => {
  let filterValue = props.value;

  const handleChange = (e) => {
    filterValue = e.target.value;
  };

  const handleSave = () => {
    const data = {
      filterId: props.filterId,
      columnId: props.columnId,
      filter: filterValue,
    };
    props.updateColumnFilter(data);
  };

  return (
    <Input
      id={`column-filter-${props.filterId}`}
      name={props.filterId}
      key={props.filterId}
      type="text"
      placeholder="Ex. = 3"
      defaultValue={props.value}
      onBlur={handleSave}
      onChange={handleChange}
      disabled={props.returningOnly}
    />
  );
};

const mapDispatchToProps = {
  updateColumnFilter,
};

export default connect(null, mapDispatchToProps)(FilterInput);
