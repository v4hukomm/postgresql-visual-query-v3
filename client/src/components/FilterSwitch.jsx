import React, { useState } from 'react';
import { connect } from 'react-redux';
import {
    CustomInput,
} from 'reactstrap';
import { updateColumn } from '../actions/queryActions';

export const FilterSwitch = (props) => {

    const [returning, setReturning] = useState(props.column.returning);
    const [returningOnly, setReturningOnly] = useState(props.column.returningOnly);

    let column = props.column;

    const handleChange = () => {
        if (props.only) {
            column.returningOnly = !column.returningOnly;
            setReturningOnly(current => !current);
            props.updateColumn(column);

        } else {
            column.returning = !column.returning;
            setReturning(current => !current);
            props.updateColumn(column);
        }
    };
    
    return (
        <CustomInput
            id={`column-${props.only ? 'returningOnly' : 'returning'}-${column.id}`}
            name={(props.only ? 'returningOnly' : 'returning')}
            key={column.id}
            type="switch"
            defaultValue={(props.only ? returningOnly : returning)}
            checked={(props.only ? returningOnly : returning)}
            disabled={props.only ? false : props.returningAll}
            onChange={handleChange}
        />
    );
};

const mapDispatchToProps = {
    updateColumn,
};

const mapStateToProps = (store) => ({
    returningAll: store.query.returning,
});

export default connect(mapStateToProps, mapDispatchToProps)(FilterSwitch);