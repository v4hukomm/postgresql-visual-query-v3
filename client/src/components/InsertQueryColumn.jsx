import React, { useState } from 'react';
import { connect } from 'react-redux';
import { 
    Button,
    CustomInput,
    Col,
    Input,
    InputGroup,
    Row, 
} from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { updateColumn, removeColumn } from '../actions/queryActions';

export const InsertQueryColumn = (props) => {

    const [returning, setReturning] = useState(props.data.returning);

    let column = props.data;
    let rowCount = Array.from({ length: props.rows }, (_, index) => index);

    const handleChange = (e) => {
        column.column_values.map((value) => {
            if (value.id === parseInt(e.target.name)) {
                value.value = e.target.value;
            }
            return value;
            });
    };

    const handleSave = () => {
        props.updateColumn(column);
    };

    const handleRemoveColumn = () => {
        props.removeColumn(props.data);
    };

    const handleSwitch = () => {
        column.returning = !column.returning;
        setReturning(current => !current);

        props.updateColumn(column);
    };

    return (
        <div>
            <h6 className='ml-4 d-inline'>{column.column_name}</h6>
            <Button
                className="ml-3"
                size="sm"
                color="danger"
                onClick={handleRemoveColumn}
                >
                <FontAwesomeIcon icon="times" />
            </Button>
            <InputGroup size="sm">
            <CustomInput
                className="mr-4 ml-4"
                type="switch"
                id={`column-returning-${column.id}`}
                key={column.id}
                name="returning"
                checked={returning}
                onChange={handleSwitch}
                label="Returning">
            </CustomInput>
            </InputGroup>
            <Col sm='1' className='mr-4 ml-4'>
                {rowCount.map(i =>
                <Row key={`column-values-row-${i}`}>
                    <Input className='mt-2 mb-2'
                    key={`column-value-${i}`}
                    name={i}
                    type="text"
                    placeholder="filter"
                    defaultValue={props.data.column_values[i].value}
                    onBlur={handleSave}
                    onChange={handleChange}
                    disabled={props.fromQuery}>
                    </Input>
                </Row>
                    )}
            </Col>
        </div>
    );
};

const mapStateToProps = (store) => ({
    fromQuery: store.query.fromQuery,
    rows: store.query.rows,
});

const mapDispatchToProps = {
    updateColumn,
    removeColumn,
};


export default connect(mapStateToProps, mapDispatchToProps)(InsertQueryColumn);