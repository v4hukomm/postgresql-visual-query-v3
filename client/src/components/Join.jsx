import React from 'react';
import { Draggable } from 'react-beautiful-dnd';
import {
  Button,
  Card,
  CardBody,
  Container,
  CustomInput,
  Form,
  FormGroup,
  Row,
  UncontrolledTooltip,
} from 'reactstrap';
import Select from 'react-select';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import _ from 'lodash';
import * as PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { removeJoin, updateJoin, updateJoinNewTable } from '../actions/queryActions';
import JoinCondition from './JoinCondition';
import { translations } from '../utils/translations';

export const Join = (props) => {
  const constructOptions = (e) => {
    let options = [];
    let newTables = [];
    let tables = [];

    props.databaseTables.map((table) => {
      const value = JSON.stringify(table);
      const option = table.table_name;
      if (table.table_schema === props.selectedSchema) {
        newTables.push({ value: value, key: `join-${props.id}-table-${table.table_name}-query-${props.queryId}`, label: option })
      };
    });

    props.tables.map((table, index) => {
      const value = JSON.stringify(table);
      const option = table.table_alias.length > 0
        ? `${table.table_name} (${table.table_alias})`
        : `${table.table_name}`;

      if (index > 0) {
        tables.push({ value: value, key: `join-${props.id}-table-${table.id}-query-${props.queryId}`, label: option })
      };
    });

    options.push({ label: translations[props.language.code].queryBuilder.joinMainTableExisting, options: tables });
    options.push({ label: translations[props.language.code].queryBuilder.joinMainTableNew, options: newTables });

    return options;
  };

  const handleTypeChange = (e) => {
    e.preventDefault();

    let join = _.cloneDeep(props.join);

    join = {
      ...join,
      type: e.target.value,
    };

    props.updateJoin(join);
  };

  const handleTableChange = (e) => {
    const value = JSON.parse(e.value);

    let join = _.cloneDeep(props.join);
    let conditions = _.cloneDeep(props.join.conditions);

    if (_.isEmpty(value.table_name)
      || (!_.isEmpty(props.join.main_table.table_name)
        && !_.isEqual(props.join.main_table.table_name, value.table_name))) {
      conditions = [];
    }

    if (value.id > 0) {
      join = {
        ...join,
        main_table: value,
        conditions,
      };

      props.updateJoin(join);
    } else {
      const newTable = constructNewTableData(value);

      let newTableWithJoin = {
        ...join,
        main_table: newTable,
        conditions,
      }

      props.updateJoinNewTable(newTableWithJoin)
    }

  };

  const handleAddCondition = () => {
    let id = 0;

    if (props.join.conditions.length > 0) {
      id = props.join.conditions[props.join.conditions.length - 1].id + 1;
    }

    const condition = {
      id,
      main_column: '',
      secondary_table: {
        table_schema: '',
        table_name: '',
        table_alias: '',
      },
      secondary_column: '',
    };

    let { join } = props;

    const conditions = _.cloneDeep(props.join.conditions);

    conditions.push(condition);

    join = {
      ...join,
      conditions,
    };

    props.updateJoin(join);
  };

  const handleRemove = () => {
    props.removeJoin(props.join);
  };

  const defaultValue = {
    table_schema: '',
    table_name: '',
    table_alias: '',
  };

  const isTableSelected = _.isEmpty(props.join.main_table.table_name);

  let firstTable;

  if (props.tables.length) {
    firstTable = props.tables[0].table_alias === ''
      ? `${props.tables[0].table_schema}.${props.tables[0].table_name}`
      : `${props.tables[0].table_alias}`;
  }
  
  const constructNewTableData = (table) => {
    const data = {
      table_schema: table.table_schema,
      table_name: table.table_name,
      table_type: table.table_type,
      table_alias: '',
    };

    let constraints = JSON.parse(JSON.stringify(props.constraints));

    constraints = constraints.filter(constraint => constraint.table_schema === data.table_schema
      && constraint.table_name === data.table_name);

    let columns = JSON.parse(JSON.stringify(props.columns));

    columns = columns.filter(column => column.table_name === data.table_name
      && column.table_schema === data.table_schema).map((column) => {
      const col = column;

      col.constraints = constraints.filter(
        constraint => _.includes(constraint.column_name, column.column_name),
      );

      delete col.table_name;
      delete col.table_schema;
      return col;
    });

    data.columns = columns;

    return data;
  };

  return (
    <div className="my-2">
      <Draggable
        draggableId={props.id}
        index={props.index}
      >
        {provided => (
          <Card
            {...provided.draggableProps}
            {...provided.dragHandleProps}
            innerRef={provided.innerRef}
          >
            <CardBody className="py-2 px-0">
              <Form>
                <Container fluid>
                  <Row>
                    <div className="col-auto d-flex">
                      <FontAwesomeIcon className="align-self-center" icon="sort" />
                    </div>
                    <div className="col-10 px-0">
                      <Row form className="mb-2 align-items-center">
                        <div className="col-auto ">
                          <FontAwesomeIcon
                            icon="link"
                            style={{ color: props.join.color }}
                          />
                        </div>
                        <div className="col-auto">
                          {props.index === 0
                            ? firstTable
                            : translations[props.language.code].queryBuilder.joinResult}
                        </div>
                        <div className="col-auto">
                          <FormGroup className="m-0">
                            <CustomInput
                              bsSize="sm"
                              type="select"
                              id={`${props.id}_join_type`}
                              onChange={handleTypeChange}
                              value={props.join.type}
                            >
                              <option value="inner">INNER JOIN</option>
                              <option value="left">LEFT JOIN</option>
                              <option value="right">RIGHT JOIN</option>
                              <option value="outer">OUTER JOIN</option>
                              <option value="cross">CROSS JOIN</option>
                            </CustomInput>
                            <UncontrolledTooltip
                              placement="top"
                              delay={{ show: 500, hide: 0 }}
                              target={`${props.id}_join_type`}
                            >
                              {translations[props.language.code].tooltips.joinType}
                            </UncontrolledTooltip>
                          </FormGroup>
                        </div>
                        <div className="col-5">
                          <FormGroup className="m-0">
                            <Select
                              id="main_table"
                              placeholder={translations[props.language.code].queryBuilder.joinMainTable}
                              onChange={handleTableChange}
                              options={constructOptions()}/>
                          </FormGroup>
                        </div>
                      </Row>
                      <Row form>
                        <div className="col-12 text-info">
                          <Button
                            className=""
                            outline
                            color="info"
                            size="sm"
                            id="addCondition"
                            disabled={isTableSelected}
                            onClick={handleAddCondition}
                          >
                            <FontAwesomeIcon icon="plus" />
                          </Button>
                          {' '}
                          {translations[props.language.code].queryBuilder.conditionH}
                        </div>
                      </Row>
                      {!_.isEmpty(props.join.conditions)
                      && (
                        <Card className="mt-2">
                          <CardBody className="py-0 px-2">
                            {props.join.conditions.map(condition => (
                              <JoinCondition
                                key={`join-${props.join.id}-condition-${condition.id}-query-${props.queryId}`}
                                condition={condition}
                                join={props.join}
                              />
                            ))}
                          </CardBody>
                        </Card>
                      )}
                    </div>
                    <div className="col-1 d-flex ml-auto pr-2 justify-content-end">
                      <FormGroup className="align-self-center m-0">
                        <Button
                          size="sm"
                          color="danger"
                          onClick={handleRemove}
                          id={`${props.id}_remove_join-${props.queryId}`}
                        >
                          <FontAwesomeIcon icon="times" />
                        </Button>
                      </FormGroup>
                    </div>
                  </Row>
                </Container>
              </Form>
            </CardBody>
          </Card>
        )}
      </Draggable>
    </div>
  );
};

Join.propTypes = {
  id: PropTypes.string,
  language: PropTypes.shape({ code: PropTypes.string }),
  join: PropTypes.shape({
    id: PropTypes.number,
    conditions: PropTypes.array,
    main_table: PropTypes.shape({ table_name: PropTypes.string }),
    color: PropTypes.string,
    type: PropTypes.string,
  }),
  updateJoin: PropTypes.func,
  removeJoin: PropTypes.func,
  tables: PropTypes.arrayOf(PropTypes.shape({
    table_alias: PropTypes.string,
    table_name: PropTypes.string,
    table_schema: PropTypes.string,
  })),
  index: PropTypes.number,
  queryId: PropTypes.number
};

const mapStateToProps = store => ({
  databaseTables: store.database.tables,
  columns: store.database.columns,
  selectedSchema: store.database.selectedSchema,
  constraints: store.database.constraints,
  lastTableId: store.query.lastTableId,
  tables: store.query.tables,
  language: store.settings.language,
});

const mapDispatchToProps = {
  updateJoin,
  removeJoin,
  updateJoinNewTable
};

export default connect(mapStateToProps, mapDispatchToProps)(Join);
