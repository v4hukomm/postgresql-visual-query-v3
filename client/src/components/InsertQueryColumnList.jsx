import React, { useState } from 'react';
import { connect } from 'react-redux';
import { Button, ButtonDropdown, DropdownToggle, DropdownItem, DropdownMenu, CustomInput, Label, Table } from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import RemoveColumnButton from './RemoveColumnButton';
import FilterSwitch from './FilterSwitch';
import ValueInput from './ValueInput';
import { addRows, removeRows, switchFromQuery, updateFromQuery, switchReturning, changeDefaultValue } from '../actions/queryActions';
import { getCorrectQueryName } from '../utils/getCorrectQueryName';

export const InsertQueryColumnList = ({
  columns, addRows, removeRows, queries, switchFromQuery,
  fromQuery, updateFromQuery, switchReturning, returning,
  language, defaultValue, changeDefaultValue,
}) => {
  const [dropDownOpen, setDropDownOpen] = useState(false);

  const handleAddRow = () => {
    addRows();
  };

  const handleRemoveRow = () => {
    removeRows();
  };

  const handleDropDown = () => {
    setDropDownOpen(current => !current);
  };

  const handleSave = (e) => {
    let subQuery;

    if (e.currentTarget.name === 'subqueryDefault') {
      subQuery = {
        subqueryId: 0,
        subquerySql: '',
      };
    } else if (e.currentTarget.name === 'subqueryId') {
      const subqueryId = +e.target.value;
      const subquerySql = queries.find(query => query.id === subqueryId).sql;
      subQuery = {
        subqueryId,
        subquerySql,
      };
    }

    updateFromQuery(subQuery);
  };

  return (
    <div className="mt-2">
      <Label>Add row</Label>
      <Button
        className="mb-1 ml-2"
        outline
        color="info"
        size="sm"
        disabled={fromQuery}
        onClick={handleAddRow}
      >
        <FontAwesomeIcon icon="plus" />
      </Button>
      <Label className="ml-2">Remove last row</Label>
      <Button
        className="mb-1 ml-2"
        outline
        color="info"
        size="sm"
        disabled={fromQuery}
        onClick={handleRemoveRow}
      >
        <FontAwesomeIcon icon="times" />
      </Button>
      <CustomInput
        inline
        className="mr-2 ml-2"
        type="switch"
        id="defaultValue"
        checked={defaultValue !== 'DEFAULT'}
        onChange={changeDefaultValue}
        label="Change DEFAULT to NULL"
      />
      <CustomInput
        inline
        className="mr-2 ml-2"
        type="switch"
        id="returning"
        checked={returning}
        onChange={switchReturning}
        label="Returning all"
      />
      <CustomInput
        inline
        className="mr-2 ml-2"
        type="switch"
        id="fromQuery"
        label="Insert from query"
        checked={fromQuery}
        onChange={switchFromQuery}
      />
      {fromQuery && (
      <ButtonDropdown
        isOpen={dropDownOpen}
        toggle={handleDropDown}
      >
        <DropdownToggle
          className="btn-sm btn-light btn-outline-secondary"
          style={{ borderColor: '#d3d8de' }}
          caret
        >
          QueryName
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem
            key="query-link-SQ"
            name="subqueryDefault"
            value=""
            onClick={handleSave}
          >
            LinkQuery
          </DropdownItem>
          {queries.map((query, index) => (
            <DropdownItem
              key={`query-${query.id}-column`}
              id={`subquerySql-${index}`}
              name="subqueryId"
              value={query.id}
              onClick={handleSave}
            >
              {getCorrectQueryName(language, query.queryName, query.id)}
            </DropdownItem>
          ))}
        </DropdownMenu>
      </ButtonDropdown>
      )}
      {!!columns.length
            && (
            <Table style={{ width: 'auto' }}>
              <thead>
                <tr>
                  <th className="border-right">Column name</th>
                  {columns.map((column) => (
                    <th className="border-right">
                      {column.table_name}
                      .
                      {column.column_name}
                      <RemoveColumnButton
                        column={column}
                      />
                    </th>
                  ))}
                </tr>
                <tr>
                  <th className="border-right">Returning</th>
                  {columns.map((column) => (
                    <td className="border-right">
                      <FilterSwitch
                        column={column}
                        only={false}
                      />
                    </td>
                  ))}
                </tr>
                <tr>
                  <th className="border-right">Returning only</th>
                  {columns.map((column) => (
                    <td className="border-right">
                      <FilterSwitch
                        column={column}
                        only
                      />
                    </td>
                  ))}
                </tr>
              </thead>
              <tbody>
                {!fromQuery && columns[0].column_values.map((_, index) => (
                  <tr>
                    <td className="text-right">{index === 0 ? 'VALUES' : ''}</td>
                    {columns.map((column) => (
                      <td>
                        <ValueInput
                          column={column}
                          index={index}
                          fromQuery={fromQuery}
                          returningOnly={column.returningOnly}
                        />
                      </td>
                    ))}
                  </tr>
                ))}
              </tbody>
            </Table>
            )}
    </div>
  );
};

const mapStateToProps = (store) => ({
  columns: store.query.columns,
  rows: store.query.rows,
  queries: store.queries,
  fromQuery: store.query.fromQuery,
  returning: store.query.returning,
  language: store.settings.language,
  defaultValue: store.query.defaultValue,
});

const mapDispatchToProps = {
  addRows,
  removeRows,
  switchFromQuery,
  updateFromQuery,
  switchReturning,
  changeDefaultValue,
};

export default connect(mapStateToProps, mapDispatchToProps)(InsertQueryColumnList);
