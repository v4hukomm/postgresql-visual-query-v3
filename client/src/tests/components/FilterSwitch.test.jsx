import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import { FilterSwitch } from '../../components/FilterSwitch';

describe('Component: FilterSwitch', () => {
    let props;
    let mockStore;
    let component;
  
    beforeEach(() => {
    props = {
        column: {
            returning: false,
            returningOnly: false,
        },
        only: false,
        updateColumn: jest.fn(),
    }
  
      mockStore = configureStore([thunk]);
    });

    test('FilterSwitch renders with default props', () => {
        component = shallow(<FilterSwitch {...props} />);
    
        expect(component).toMatchSnapshot();
    });

    test('FilterSwitch handleSave calls updateColumn once', () => {
        component = shallow(<FilterSwitch {...props} />).simulate('change');

        expect(props.updateColumn.mock.calls.length).toBe(1);
    });

    test('FilterSwitch handleSave calls updateColumn once if only is true', () => {

        props = {
            column: {
                returning: false,
                returningOnly: false,
            },
            only: true,
            updateColumn: jest.fn(),
        }

        component = shallow(<FilterSwitch {...props} />).simulate('change');

        expect(props.updateColumn.mock.calls.length).toBe(1);
    });
});