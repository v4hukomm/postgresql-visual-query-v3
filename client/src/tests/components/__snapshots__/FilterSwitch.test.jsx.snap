// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: FilterInput FilterSwitch renders with default props 1`] = `ShallowWrapper {}`;

exports[`Component: FilterSwitch FilterSwitch renders with default props 1`] = `ShallowWrapper {}`;
